src/core/eval.ml

```diff
@@ -77,38 +79,55 @@ let snf : (term -> term) -> (term -> term) = fun whnf ->
     | TRef(_)     -> assert false
   in snf
 
-(** Configuration of the reduction engine. *)
-type config =
-  { context : ctxt (** Context of the reduction used for generating metas. *)
-  ; defmap : term VarMap.t (** Variable definitions. *)
-  ; rewrite : bool (** Use user-defined rewrite rules. *)
-  ; problem : problem (** Generated metavariables. *) }
-
-(*let pp_defmap = D.map VarMap.iter pp_var " ≔ " pp_term "; "*)
+type rw_tag = [ `NoBeta | `NoRw | `NoExpand ]
 
-let cfg_of_ctx : ctxt -> bool -> config = fun context rewrite ->
-  {context; defmap = Ctxt.to_map context; rewrite; problem = new_problem()}
-
-let unfold_cfg : config -> term -> term = fun c a ->
-  let a = unfold a in
-  match a with
-  | Vari x ->
-    begin match VarMap.find_opt x c.defmap with
-      | None -> a
-      | Some v -> unfold v
-    end
-  | _ -> a
+(** Configuration of the reduction engine. *)
+module Config = struct
+
+  type t =
+    { context : ctxt
+    (** Context of the reduction used for generating metas. *)
+    ; defmap : term VarMap.t (** Variable definitions. *)
```

Comment:
To avoid possible confusions, I suggest to rename defmap into varmap.

Reply:
Btw, this `defmap` is only optimisation, isn't it?

Reply:
Probably.